### PR TITLE
Readme updates

### PR DESCRIPTION
**Pull request by AnthonySteele**
better wordings, fixes to code samples and added section headings

open questions:
- what does "IStatsDPublisher.MarkEvent" do?

- In the context of an IoC container, can `IStatsDPublisher` be a singleton, must it be transient or some other lifestyle?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,30 +4,28 @@
...
 * statsd metrics formatter
 * UDP client handling
 
-####
+#### Publishing statistics
+
+`IStatsDPublisher` is the interface that you will use in most circumstances. With this you can `Increment` or `Decrement` an event, and send values for a `Gauge` or `Timing`.
 
-`IStatsDPublisher` is the interface that you will use in most circumstances. Use an instance of this in order to send events, timers and gauges.
+The concrete class that implements `IStatsDPublisher` is `StatsDImmediatePublisher`. For the constructor parameters, you will need the statsd server host name. You can change the standard port (8125). You can also append a prefix to all stats. These values often come from configuration as the host name and/or prefix may vary between test and production environments.
```
Comment (petemounce) [inline README.md]: "append a prefix" sounds wrong. Prepend. We should add the ability to have both prefix and suffix, what with the difficulty we have with inconsistently named metrics...

File: README.md [hunk 2]
```diff
@@ -36,25 +34,26 @@ example of Ioc in NInject for statsd publisher with values from config:
...
+		stats.Increment("DoSomething-Attempt");
 		var stopWatch = Stopwatch.StartNew();
-
         var success = DoSomething();
 
 		stopWatch.Stop();
 		if (success)
         {
-			stats.Timing("DoSomething.Success", stopWatch.Elapsed);
+			stats.Timing("DoSomething-Success", stopWatch.Elapsed);
```
Comment (petemounce) [inline README.md]: I'd keep the `.` here. Segments with more than one piece of logical data are less flexible.

File: README.md [hunk 3]
```diff
@@ -4,30 +4,28 @@
 
 ### TL;DR
 
-We use this within our components to publish [statsd](http://github.com/etsy/statsd) metrics.
+We use this within our components to publish [statsd](http://github.com/etsy/statsd) metrics from .Net code.
```
Comment (petemounce) [inline README.md]: nit: `.NET`.

File: README.md (hunk 1 above)
Comment (AnthonySteele) [inline README.md]: Good spot, I have fixed it.

If you want to add another config option for a stat suffix then you would need another constructor parameter.I'm not a great fan of injecting primitive types such as these `string` and `int` using an IoC container. It's a less-used feature of the various containers, meaning that support and syntax is going to vary between different containers; developer familiarity will be lower etc. 

So when you find yourself doing more of it, there may be a better way. i.e. making a type to hold all these values, so that programmer and IoC container have a unique type to work with. It also allows for sensible defaults.
e.g. 
```csharp
public class StatsDSettings
{
   public int Port { get; set; }
   public string Host { get; set; }
  // etc
}

public class StatsDPublisher
{
   public StatsDPublisher(StatsDSettings settings)
   {
   // etc
  }
}

```

**Comment by AnthonySteele** [comment]
Regarding future work on more configuration options for prefixes ans suffixes:

If you want to add another config option for a stat suffix then you would need another constructor parameter.I'm not a great fan of injecting primitive types such as these `string` and `int` using an IoC container. It's a less-used feature of the various containers, meaning that support and syntax is going to vary between different containers; developer familiarity will be lower etc. 

So when you find yourself doing more of it, rather than "add more parameters to the method", there may be a better way. i.e. making a type to hold all these values, so that programmer and IoC container have a unique type to work with. It also allows for sensible defaults.
e.g. 
```csharp
public class StatsDSettings
{
   public int Port { get; set; }
   public string Host { get; set; }
  // etc
}

public class StatsDPublisher
{
   public StatsDPublisher(StatsDSettings settings)
   {
   // etc
  }
}

```

**Comment by AnthonySteele** [comment]
Can we merge this? It may not be perfect, but I think it's an improvement.

**Comment by petemounce** [comment]
Yes, totally happy to iterate!
